wordpress-mobile/WordPress-Android
# [Blogging prompts v1] Implement introduction dialog analytics events

Fixes #16521

To test:
- Enable blogging prompts feature flag
- Open app logged in and with a site selected
- Click on the blogging prompts onboarding notification OR in the "Learn more" button
- Blogging prompts onboarding dialog should be seen
  - `"blogging_prompts_introduction_modal_viewed"` event should be triggered when the dialog is shown
  - `"blogging_prompts_introduction_modal_dismissed"` event should be triggered when user taps on "x" button or back button
  - `"blogging_prompts_introduction_modal_try_it_now_tapped"` event should be triggered when user taps on "Try it now" button
  - `"blogging_prompts_introduction_modal_remind_me_tapped"` event should be triggered when user taps on "Remind me" button
  - `"blogging_prompts_introduction_modal_got_it_tapped"` event should be triggered when user taps on "Got it" button

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests

3. What automated tests I added (or what prevented me from doing so)
Updated `BloggingPromptsOnboardingViewModelTest.kt` and implemented `FeatureIntroductionViewModelTest.kt`.

PR submission checklist:
